src/api/app/controllers/webui/packages/badge_controller.rb

```diff
@@ -3,32 +3,34 @@ module Packages
     class BadgeController < Packages::MainController
       before_action :set_project
       before_action :set_package
+      RED = '#e05d44'.freeze
+      GREEN = '#4c1'.freeze
 
       def show
         results = @package.buildresult(@project, false, true).results[@package.name]
         results = results.select { |r| r.architecture == params[:architecture] } if params[:architecture]
         results = results.select { |r| r.repository == params[:repository] } if params[:repository]
-        send_data(create_badge(results), type: 'image/svg+xml', disposition: 'inline')
+        send_data(create_badge(results, params[:type]), type: 'image/svg+xml', disposition: 'inline')
       end
 
       private
 
-      def create_badge(results)
+      def create_badge(results, type)
```

Comment:
This looks like an object by itself (`Badge.new(results, type`) with it's own tests and so on... :)